docs/wiki/development/coding-guidelines-cpp.md

```diff
@@ -80,21 +80,21 @@ All classes should strictly maintain separation between declarations and definit
 
 ### Protected/Private Members/Methods
 
-All protected/private members and methods should be prefixed with a single `_` to provide easy identification that the value is not in the public scope.
+All protected/private members and methods should be suffixed with a single `_` to provide easy identification that the value is not in the public scope.
 
 Example:
 
 ```c++
 //test_class.hpp
 class test_class {
 public:
-    test_class(uint32_t init_);
+    test_class(uint32_t init);
 protected:
-    uint32_t _val;
+    uint32_t val_;
```

Comment:
Same as my comment above but inverted.

Reply:
I see a problem with this approach - while docs indicate that we should use underscore with lowercase text identifier, like `_val`, this could cause annoying problems if something like `_Val` slips through review.

Imagine following case:
```
// something.h
class something {
  public:
    void do_something_with_val();
  private:
    int _Val;
};

// something.cpp
// #include "something.h"
#include <vector> // doesn't define _Val, but compiler devs can change it

#define COMPILER_UPDATED // let's say that COMPILER_UPDATED means, that vector include has changed, and including it results in defining _Val as false
#ifdef COMPILER_UPDATED
#define _Val false
#endif

void something::do_something_with_val()
{
  if (_Val != 1024) {
     // do_something_important
  }
}

int main()
{
    something s;
    // some internal operations that may change _Val to 1024
    s.do_something_with_val();
}
```

If there's a bigger review, then it's really easy to let a private variable like `_Val` to slip through, into the code. While it's unlikely case, it always can happen - compiler devs have the right to define anything they want, within the limits described by C++ standard.

Reply:
Only arguments are suffixed. Members are without prefixes or suffixes

Reply:
Now it should be fine.